frame/nomination-pools/src/migration.rs

```diff
@@ -724,4 +724,106 @@ pub mod v5 {
 			Ok(())
 		}
 	}
+
+	/// This migration summarizes and initializes the TotalValueLocked StorageValue for all Pools.
+	pub struct MigrateToV6<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV6<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
```

Comment:
Please use `VersionedRuntimeUpgrade`

Reply:
I introduced `VersionedRuntimeUpgrade` but I'm not 100% sure if it's correctly done.

Reply:
Looks good from a cursory glance. 

We'll test it in more detail with [`try-runtime-cli`](https://github.com/paritytech/try-runtime-cli) once the rest of the PR is ready :)